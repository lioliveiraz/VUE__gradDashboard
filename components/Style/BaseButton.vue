<template>
  <button class="g-button-base" @click="handleClick">
    {{ value }}
  </button>
</template>

<script>
export default {
  props: {
    handleClick: { type: Function, required: false },
    value: String,
  },
};
</script>

<style>
</style>