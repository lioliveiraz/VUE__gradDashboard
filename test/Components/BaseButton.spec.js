import BaseButton from '../../components/Style/BaseButton.vue';
import { shallowMount } from '@vue/test-utils';

describe('<BaseButton>', () => {
    let wrapper, handleClick, value;
    handleClick = jest.fn();
    value = 'value';
    beforeEach(async () => {

        wrapper = await shallowMount(BaseButton, {
            propsData: { handleClick, value }, mocks: {
                $t: (msg) => msg,

            }
        });
    });
    it('should render correctly', () => {

        expect(wrapper.find("g-base-btn-blue")).toBeTruthy();
        expect(wrapper).toMatchSnapshot();

    });
    it('should receive props correctly', () => {
        expect(wrapper.props()).toEqual({ handleClick, value });

    });
});