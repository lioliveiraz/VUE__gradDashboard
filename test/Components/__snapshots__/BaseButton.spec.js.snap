// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseButton> should render correctly 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
